components/breadcrumbs/test/breadcrumbs.test.js

```diff
@@ -0,0 +1,161 @@
+
+describe('d2l-breadcrumbs', () => {
+	describe('basic attribute tests', () => {
+		let elem;
+		beforeEach(async() => {
+			elem = await fixture(html`
+				<d2l-breadcrumbs>
+					<d2l-breadcrumb href="#" text="Basic Item 1"></d2l-breadcrumb>
+					<d2l-breadcrumb href="#" text="Basic Item 2"></d2l-breadcrumb>
+					<d2l-breadcrumb href="#" text="Basic Item 3" aria-label="Aria for Item 3"></d2l-breadcrumb>
+				</d2l-breadcrumbs>
+			`);
+		});
+
+		describe('attribute reflection', () => {
+			it('should reflect "compact" property to attribute', async() => {
+				elem.compact = true;
+				await elem.updateComplete;
+				expect(elem.hasAttribute('compact')).to.be.true;
+			});
+
+			it('should reflect "compact" property to children', async() => {
+				elem.compact = true;
+				await elem.updateComplete;
+				expect(elem.shadowRoot.querySelector('slot').assignedNodes().filter(
+					item => item.nodeName === 'D2L-BREADCRUMB' &&
+				item.hasAttribute('compact')
+				).length).to.eq(3);
+			});
+		});
+
+		describe('constructor', () => {
+			it('should construct', () => {
+				runConstructor('d2l-breadcrumbs');
+			});
+		});
+
+		describe('default property values', () => {
+			it('should default "compact" property to "false" when unset', () => {
+				expect(elem.compact).to.be.false;
+			});
+		});
+	});
+
+	describe('nested children attributes', () => {
+		let elem;
+		const flatten = nodes => {
+			let flatResult = [...nodes];
+			nodes.forEach(node => {
+				if (node.children) {
+					flatResult = flatResult.concat(flatten([...node.children]));
+				}
+			});
+			return flatResult;
+		};
+
+		beforeEach(async() => {
+			elem = await fixture(html`
+				<d2l-breadcrumbs>
+					<d2l-breadcrumb href="#" text="Basic Item 1"></d2l-breadcrumb>
+					<div>
+						<d2l-breadcrumb href="#" text="Basic Item 2"></d2l-breadcrumb>
+					</div>
+					<div>
+						<div>
+							<d2l-breadcrumb href="#" text="Basic Item 3" aria-label="Aria for Item 3"></d2l-breadcrumb>
+						</div>
+					</div>
+				</d2l-breadcrumbs>
+			`);
+		});
+
+		describe('attribute reflection', () => {
+			it('should reflect "compact" property to nested children', async() => {
+				elem.compact = true;
+				await elem.updateComplete;
+				const assignedNodes = elem.shadowRoot.querySelector('slot').assignedNodes();
+				expect(flatten(assignedNodes).filter(item => item.nodeName === 'D2L-BREADCRUMB' && item.hasAttribute('compact')).length).to.eq(3);
+			});
+		});
+	});
+
+	describe('basic', () => {
+		let list;
+		let items;
+		before(async() => {
+			list = await fixture(html`
+				<d2l-breadcrumbs>
+					<d2l-breadcrumb href="#" text="Basic Item 1"></d2l-breadcrumb>
+					<d2l-breadcrumb href="#" text="Basic Item 2"></d2l-breadcrumb>
+					<d2l-breadcrumb href="#" text="Basic Item 3" aria-label="Aria for Item 3"></d2l-breadcrumb>
+				</d2l-breadcrumbs>
+			`);
+			items = list.querySelectorAll('d2l-breadcrumb:not([hidden])');
+		});
+
+		it('should display three items', () => {
```

Comment:
I find the visual-diff tests address this, and much more... making many such tests redundant.  I think even some of the property reflection is captured by the visual-diff test.  In some cases, the property reflection is simply a means to achieve some visual presentation, which is really what we're interested in.  Some of these tests were typical before we had visual-diff testing as a tool available to us.

Reply:
@dbatiste do visual-diff tests also compare against multiple browsers? I think it already caught some of the issues with edge which I wouldn't have caught on my own

Reply:
Currently we have it configured to cover Chromium browsers.  So, we can make a call on what's interesting to test cross-browser.  Many of the unit tests from previous Polymer components don't survive the migration.

Reply:
So do you suggest removing the these tests for now? or should it be done at a later stage when we are more confident with cross-browser visual diffs?

Reply:
Yeah I agree with Dave here... I'd probably remove everything except the constructor tests and maybe the default value ones. Basically if someone later mucked around with the code, we want to make sure that we catch things that the visual diff tests will not, but without going overboard.

Reply:
Re cross-browser visual-diff: I think it's possible to launch experimental FF with Puppeteer.  It would be a big deal though, as it would significantly increase our build times, and undoubtedly introduce some build instability.  I think there is some mention of having Puppeteer drive headless Safari too, but to my knowledge this is merely an idea.

Reply:
Removed ✔️